tools/protoxform/merge_active_shadow.py

```diff
@@ -37,8 +37,23 @@ def AdjustReservedRange(target_proto, previous_reserved_range, skip_reserved_num
       target_proto.reserved_range.add().MergeFrom(rr)
 
 
+# Add dependencies for envoy.annotations.disallowed_by_default
+def AddDeprecationDependencies(target_proto_dependencies, proto_field, is_enum):
```

Comment:
Why is this part of this PR?

Reply:
This part is needed when the shadow proto brings a deprecated field that uses `envoy.annotations.disallowed_by_default` and the original proto does include the import for `envoy/annotations/deprecation.proto`.

Reply:
Makes sense. What about the `struct.proto` below?

Reply:
`struct.proto` was needed when the merge shadow added something like
```
  map<string, google.protobuf.Struct> hidden_envoy_deprecated_per_filter_config = 12
      [deprecated = true];
```
e.g. from `api/envoy/config/route/v3/route_components.proto` to `generated_api_shadow/envoy/config/route/v3/route_components.proto` and struct was not already in the imports.